src/alchemlyb/preprocessing/subsampling.py

```diff
@@ -97,14 +96,10 @@ def statistical_inefficiency(df, series=None, lower=None, upper=None, step=None)
 
         # calculate statistical inefficiency of series
         statinef  = statisticalInefficiency(series)
-
-        # we round up
-        statinef = int(np.rint(statinef))
-
-        # subsample according to statistical inefficiency
-        series = series.iloc[::statinef]
-
-        df = df.loc[series.index]
+        
+        #use the subsampleCorrelatedData function to get the subsample index
```

Comment:
update the docs: make clear that intervals may be not uniform any more.

Reply:
yes, will add docs